fiftyone/operators/executor.py

```diff
@@ -443,11 +446,34 @@ def dataset(self):
         """The :class:`fiftyone.core.dataset.Dataset` being operated on."""
         if self._dataset is not None:
             return self._dataset
-
-        dataset_name = self.request_params.get("dataset_name", None)
-
-        self._dataset = fod.load_dataset(dataset_name)
-        self._dataset.reload()
+        dataset_name = None
+        # since the dataset may have been renamed, if we have the
+        # dataset_id, get the current dataset_name from the db
+        if self.request_params.get("dataset_id", None) is not None:
```

Comment:
How about a utility function like `load_dataset(id=None, name=None)` that loads the dataset by ID if provided, else name?

The idea of loading a dataset by ID is likely useful elsewhere